conda_build/source.py

```diff
@@ -1109,6 +1115,25 @@ def provide(metadata):
                 if not isdir(src_dir):
                     os.makedirs(src_dir)
 
+            for hash_type in CONTENT_HASH_KEYS:
+                if hash_type in source_dict:
+                    expected_content_hash = source_dict[hash_type]
+                    if expected_content_hash in (None, ""):
+                        raise ValueError(
+                            f"Empty {hash_type} hash provided for source item #{idx}"
+                        )
+                    algorithm = hash_type[len("content_") :]
+                    obtained_content_hash = compute_content_hash(
+                        src_dir,
+                        algorithm,
+                        skip=ensure_list(source_dict.get("content_hash_skip") or ()),
+                    )
+                    if expected_content_hash != obtained_content_hash:
+                        raise RuntimeError(
+                            f"{hash_type} mismatch in source item #{idx}: "
```

Comment:
Is it not possible to list the source url here?

Reply:
There are too many possibilities: `git_url`, `url`, `path`, `svn`... And we would have to adjust that code to cover for the potential URL origin. I think it's one of those rare cases where it pays to be a bit lazy 😬

Reply:
I wonder how often we have ever added url types, but ok fair enough.